vxsandbox/resources/tests/test_utils.py

```diff
@@ -1 +1,288 @@
+        resources = self.mk_resources(config={
+            'tst1': {'cls': self.class_name(RecordingResource)},
+            'tst2': {'cls': self.class_name(RecordingResource)},
+        })
+        resources.validate_config()
+        self.assert_recording_resources(resources, {
+            'tst1': {'setup_calls': 0},
+            'tst2': {'setup_calls': 0},
+        })
+        resources.setup_resources()
+        self.assert_recording_resources(resources, {
+            'tst1': {'setup_calls': 1},
+            'tst2': {'setup_calls': 1},
+
+        })
+
+    def test_teardown_resources(self):
+        resources = self.mk_resources(config={
+            'tst1': {'cls': self.class_name(RecordingResource)},
+            'tst2': {'cls': self.class_name(RecordingResource)},
+        })
+        resources.validate_config()
+        self.assert_recording_resources(resources, {
+            'tst1': {'teardown_calls': 0},
+            'tst2': {'teardown_calls': 0},
+        })
+        resources.teardown_resources()
+        self.assert_recording_resources(resources, {
+            'tst1': {'teardown_calls': 1},
+            'tst2': {'teardown_calls': 1},
+        })
+
+
+class CommandHandler(object):
+    def __init__(self, name, test_case, result=None):
+        self._name = name
+        self._test_case = test_case
+        self._calls = []
+        self._result = result or object()
+
+    def assert_called(self, api, cmd, result_d):
+        self._test_case.assertEqual(self._calls, [(api, cmd)])
+        if isinstance(self._result, Deferred):
+            self._test_case.assertEqual(result_d, self._result)
+        else:
+            self._test_case.assertTrue(result_d.called)
+            self._test_case.assertEqual(result_d.result, self._result)
+
+    def __call__(self, api, cmd):
+        self._calls.append((api, cmd))
+        return self._result
+
+
+class TestSandboxResource(VumiTestCase):
+    def mk_resource(self, name, app_worker=None, config=None, handler=None):
+        app_worker = app_worker or object()
+        config = config or {}
+        resource = SandboxResource(name, app_worker, config)
+        if handler is not None:
+            setattr(resource, handler._name, handler)
+        return resource
+
+    def assert_recording_api(self, api, logs=None, deaths=0):
+        logs = logs or []
+        self.assertEqual(api.logs, logs)
+        self.assertEqual(api.deaths, deaths)
+
+    def test_create(self):
+        dummy_worker = object()
+        resource = SandboxResource(
+            'noir', dummy_worker, {'colour': 'black'})
+        self.assertEqual(resource.name, 'noir')
+        self.assertEqual(resource.app_worker, dummy_worker)
+        self.assertEqual(resource.config, {'colour': 'black'})
+
+    def test_setup(self):
+        resource = self.mk_resource('test')
+        resource.setup()
+
+    def test_teardown(self):
+        resource = self.mk_resource('test')
+        resource.teardown()
```

Comment:
These are just checking that the base implementations of ``setup()`` and
``teardown()`` exist and can be called with the expected (i.e. no)
arguments. I was sort of wondering how to assert that they do nothing, but
didn't come up with any good ideas.